### DISCUSSION TITLE
Unable to get Mixed mode debugging working in python 3.9.6 + pythonnet 2.5.2

### DISCUSSION THREAD

**Discussion by guytaylor** [pythonnet/pythonnet#1741] [Q&A]
### Environment

-   Pythonnet version: 2.5.2
-   Python version: 3.9.6
-   Operating System: windows 10
-   .NET Runtime: 4.6.2

### Details

-   Describe what you were trying to get done.

   I am trying to get mixed mode debugging working. I have a c# app which calls python and I would like to be able to debug into the python code.

I've followed the instruction here: https://github.com/pythonnet/pythonnet/wiki/Various-debugging-scenarios-of-embedded-CPython
and worked my way through this thread: https://github.com/pythonnet/pythonnet/issues/332

This is the C# app:
    
```
public class Program
    {
        public static void EnableDebugging()
        {
            Console.WriteLine("waiting for .NET debugger to attach");
            while (!Debugger.IsAttached)
            {
                Thread.Sleep(100);
            }
            Console.WriteLine(".NET debugger is attached");
        }

        public static void Main(string[] args)
        {
            string pythonHome = @"C:\Python39";
            string pythonPath = pythonHome + @"\DLLs;" + pythonHome + @"\Lib;" + pythonHome + @"\Lib\site-packages";
            pythonPath += ";" + @"C:\Users\gtaylor\Downloads\EmbeddedPythonDebugging-master\EmbeddedPythonDebugging\bin\Debug";
            string path = pythonHome + ";" + Environment.GetEnvironmentVariable("PATH", EnvironmentVariableTarget.Machine);
            Environment.SetEnvironmentVariable("PATH", path, EnvironmentVariableTarget.Process);
            Environment.SetEnvironmentVariable("PYTHONHOME", pythonHome, EnvironmentVariableTarget.Process);
            Environment.SetEnvironmentVariable("PYTHONPATH ", pythonPath, EnvironmentVariableTarget.Process);

            PythonEngine.PythonHome = pythonHome;
            PythonEngine.PythonPath = pythonPath;

            PythonEngine.Initialize();
            
            EnableDebugging();

            using (Py.GIL())
            {
                PythonEngine.ImportModule("testmodule");
            }
        }
    }
```

In the project conditional compilation symbols are: `PYTHON3;PYTHON39;UCS2`

The python code in testmodule.py is:

```
import time
while True:
    print ("Hello world..") # set a break point here
```

These are steps I take to run it:

1. Build
2. From a command prompt, run the app
    - I get the prompt `waiting for .net debugger to attach`
3. I attach the debugger like this:

<img width="336" alt="pic1" src="https://user-images.githubusercontent.com/34499540/159637451-e17bfc9a-f199-4758-a059-f7f174113359.png">

This is what it looks like after attaching:

<img width="743" alt="debug" src="https://user-images.githubusercontent.com/34499540/159637420-7c8ca767-b076-49b0-a895-27fd0be9af76.png">

4. The app continues, and you can see the output:
<img width="691" alt="console" src="https://user-images.githubusercontent.com/34499540/159606181-3d1edbbc-eabc-411c-9c39-1bc1265c3ba6.png">

5. My breakpoint is never hit. I have tried this with both the 32bit and 64bit version. (I need 64 bit for my application)
<img width="1003" alt="pic3break" src="https://user-images.githubusercontent.com/34499540/159606260-6245a80d-bbac-4440-a1e3-e0d122bd2023.png">

This is what my modules look like:
<img width="1666" alt="pics2mods" src="https://user-images.githubusercontent.com/34499540/159606330-e2943605-28f3-4a9f-9ad5-b56d57952662.png">

**Comment by guytaylor** [comment]
@filmor, do you know if this _should_ work? I am running VS 2019 16.9.9
or is debugging of 3.9.6 not supported in that version?

**Comment by 731315163** [comment]
how do debug in vscode ?
